templates/filter/filter.js

```diff
@@ -0,0 +1,60 @@
+import ffetch from '../../scripts/ffetch.js';
+
+const alphaArr = ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z', '#'];
+
+function dataMapLookup(value, dataMap) {
+  const foundValue = dataMap.find((item) => item.key === value);
+  return foundValue ? foundValue.value : value;
+}
+
+function createDropdownOption(value, dataMap) {
+  const option = document.createElement('option');
+  option.innerText = dataMapLookup(value, dataMap);
+  option.setAttribute('value', value);
+  return option;
+}
+
+// Function to get unique values from an array of keys for each object
+function getUniqueValuesByKeys(data, key) {
+  const uniqueValues = [];
+
+  data.forEach((obj) => {
+    const value = obj[key];
+
+    // Check if the value is not in the uniqueValues array for this key
+    if (!uniqueValues.includes(value)) {
+      uniqueValues.push(value);
+    }
+  });
+
+  return uniqueValues;
+}
+
+// eslint-disable-next-line import/prefer-default-export
+export async function loadDelayed(main) {
+  const dropdowns = main.querySelectorAll('select[data-populated="false"]');
+  const dataSource = dropdowns[0].getAttribute('data-source');
+  const year = dropdowns[0].getAttribute('data-year');
+  const dataSheet = `/data-source/${dataSource}/${dataSource}-data.json`;
+  const dataMapSheet = `/data-source/${dataSource}/data-mapping.json`;
+
+  const data = await ffetch(dataSheet).sheet(year).all();
+  const dataMap = await ffetch(dataMapSheet).sheet(year).all();
```

Comment:
Is this happening in addition to what the block(s) are doing? Will all the blocks in this PR potentially be on the same page? Lastly, are the HTTP requests that they're all making the same?

I ask because I'm just trying to think through how we could further optimize all these `ffetch` calls.

One thought that might help us eek out a bit more performance is to do _all_ the HTTP requests in one place. For example, something like this:

* The blocks don't actually call `ffetch`, they just create some kind of placeholder content.
* In the template's `loadLazy()`, we run the queries we need using the `Promise.all()` strategy.
* After we have the data, the template puts the data in `data-` attribute(s) on the blocks.
* The blocks are listening for attribute changes, and when they get the data they replace their placeholders with actual content.

I did something similar in the [article-cards](https://github.com/hlxsites/channelco-crn-com/blob/main/blocks/article-cards/article-cards.js), so we could use that as a template if needed.

We'd potentially only need this if the other optimizations aren't enough. I'd be happy to assist or answer any questions, just feel free to let me know.